providers/redis/redis.go

```diff
@@ -287,8 +303,62 @@ func (p *RedisProvider) GetState(ctx context.Context, instanceID string) (provid
 		return providers.ServiceState(""), "", fmt.Errorf("Invalid response from AWS: status is missing for %s", replicationGroupID)
 	}
 
-	message := p.getMessage(ctx, replicationGroup)
+	if timeOut != "" {
```

Comment:
Could we handle the parsing of the time before reaching this method and pass a `*time.Time` here? Would make the method a little simpler and reduce its overall number of responsibilities.

Reply:
Moved to the broker code